libtransmission/torrent.cc

```diff
@@ -1903,6 +1903,29 @@ void tr_torrentSetLabels(tr_torrent* tor, tr_labels_t&& labels)
 ****
 ***/
 
+void tr_torrentSetGroup(tr_torrent* tor, std::string_view group)
+{
+    TR_ASSERT(tr_isTorrent(tor));
+    auto const lock = tor->unique_lock();
+
+    tor->group = std::string{ group };
+    tr_bandwidth_group* bw_group = tr_bandwidthGroupFind(tor->session, group);
+    if (bw_group == nullptr)
+    {
+        tor->bandwidth->setParent(tor->session->bandwidth);
+    }
+    else
+    {
+        tor->bandwidth->setParent(bw_group->bandwidth);
+    }
```

Comment:
This sets `tor->group()` here even if `tr_bandwidthGroupFind()` returns nullptr. Is that intentional?

Reply:
Yes. This happens when a name is empty, and then the torrent belongs to the session bandwidth object.